ps_viewedproduct.php

```diff
@@ -192,7 +192,11 @@ public function renderWidget($hookName = null, array $configuration = [])
 
     public function getWidgetVariables($hookName = null, array $configuration = [])
     {
-        $products = $this->getViewedProducts($configuration['product']['id_product']);
+        if ('product' === $this->context->controller->php_self) {
+            $products = $this->getViewedProducts($this->context->controller->getProduct()->id);
+        } else {
+            $products = $this->getViewedProducts(false);
+        }
```

Comment:
For this check, I was also using the configuration variable.

$id_product = isset($configuration['product']['id_product']) ? $configuration['product']['id_product'] : false;$products = $this->getViewedProducts($id_product);